cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/DeduplicateAggregationDefinition.java

```diff
@@ -0,0 +1,176 @@
+     * only one record from the group is extracted.
+     * Any existing dedup expression list is overwritten.
+     * @param dedupExpressions {@link Expression}s to deduplicate on.
+     * @return a {@link Builder} with the currently built {@link DeduplicateAggregationDefinition}.
+     */
+    public Builder dedupOn(Expression... dedupExpressions) {
+      return dedupOn(Arrays.asList(dedupExpressions));
+    }
+
+    /**
+     * Sets the list of expressions to select to the specified list of expressions.
+     * Any existing list of select expressions is overwritten.
+     * @param selectExpressions list of {@link Expression}s to select.
+     * @return a {@link Builder} with the currently built {@link DeduplicateAggregationDefinition}.
+     */
+    public Builder select(List<Expression> selectExpressions) {
+      this.selectExpressions = selectExpressions;
+      return this;
+    }
+
+    /**
+     * Sets the list of expressions to select to the specified expressions.
+     * Any existing list of select expressions is overwritten.
+     * @param selectExpressions {@link Expression}s to select.
+     * @return a {@link Builder} with the currently built {@link DeduplicateAggregationDefinition}.
+     */
+    public Builder select(Expression... selectExpressions) {
+      return select(Arrays.asList(selectExpressions));
+    }
+
+    /**
+     * Adds the specified expression with the specified filter function to the list of filter expressions.
+     * This function is applied on this expression for all the records for which the values of the dedup expressions
+     * specified in <code>dedupOn</code> is the same, and the resultant row is extracted by the dedup operation.
+     * @param filterExpression an {@link Expression} on which the filter function will be applied.
+     * @param filterFunction a {@link FilterFunction} to be applied to the filter expression.
+     * @return a {@link Builder} with the currently built {@link DeduplicateAggregationDefinition}.
+     */
+    public Builder filterDuplicatesBy(Expression filterExpression, FilterFunction filterFunction) {
+      this.filterExpressions.add(new SimpleEntry<>(filterExpression, filterFunction));
+      return this;
+    }
+
+    /**
+     * Sets the list of filter expressions to the specified list of filter functions and filter expressions.
+     * The filter function is applied to the respective filter expression for all the records for which the values
+     * of the dedup expressions specified in <code>dedupOn</code> is the same, and the resultant row is extracted
+     * by the dedup operation.
+     * @param filterExpressions a {@link List} of {@link SimpleEntry} pairs of
+     *                          an {@link Expression} and a {@link FilterFunction}.
+     * @return a {@link Builder} with the currently built {@link DeduplicateAggregationDefinition}.
+     */
+    public Builder filterDuplicatesBy(List<SimpleEntry<Expression, FilterFunction>> filterExpressions) {
```

Comment:
please use interface instead of implementation as the parameter type (e.g. `List<? extends Map.Entry<Expression, FilterFunction>>`)

Reply:
This is done, PTAL.